scripts/fetch_ci_data.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import csv
+import json
+import os
+import sys
+import urllib.error
+import urllib.request
+from datetime import datetime, timedelta
+from pathlib import Path
+
+
+def get_ci_for_interval(start_time, delta, postcode) -> tuple[str, float]:
+    """
+    Retrieves carbon intensity forecast data for a specified time interval and postcode.
+
+    Args:
+        start_time (datetime): The start time of the interval (UTC).
+        delta (timedelta): The duration of the interval.
+        postcode (str): The UK postcode to query carbon intensity for, e.g. EC1A. This must be only the outward postcode (the first half of the full postcode).
+
+    Returns:
+        tuple[str, float]: A tuple containing the formatted start time (as a string) and the carbon intensity forecast (in gCO2/kWh).
+
+    Raises:
+        SystemExit: If the HTTP request to the API fails.
+    """
+    dateformat = "%Y-%m-%dT%H:%MZ"
+    end_time = start_time + delta
+    start_time_url = start_time.strftime(dateformat)
+    end_time_url = end_time.strftime(dateformat)
+
+    # Query carbonintensity to get the JSON information
+    ci_url = f"https://api.carbonintensity.org.uk/regional/intensity/{start_time_url}/{end_time_url}/postcode/{postcode}"
+    ci_response = None
+    try:
+        ci_response = urllib.request.urlopen(ci_url)
+    except urllib.error.HTTPError as e:
+        print(e)
+        sys.exit(1)
```

Comment:
Depending on the type/frequency of errors we observe in practice, it might be easier to add rate limiting and retry logic here, rather than rerunning the cron job.

I’ve also recently heard about [`tenacity`](https://tenacity.readthedocs.io/en/latest/); but that’s probably more than we need here.

Reply:
`tenacity` looks very cool!

I've not come across any errors in developing and testing out the code so far, so the error rate isn't high. I think it's probably best to wait and see what errors we observe in practice, and then figure out the best way to handle them.